SpectralLineRequest.cc

```diff
@@ -0,0 +1,163 @@
+#include "SpectralLineRequest.h"
+
+#include <curl/curl.h>
+
+#include <cstdlib>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "Table/Columns.h"
+
+using namespace carta;
+
+#define REST_FREQUENCY_COLUMN_INDEX 2
+
+const std::string SpectralLineRequest::SplatalogueURL =
+    "https://www.cv.nrao.edu/php/splat/"
+    "c_export.php?&sid%5B%5D=&data_version=v3.0&lill=on&displayJPL=displayJPL&displayCDMS=displayCDMS&displayLovas=displayLovas&"
+    "displaySLAIM=displaySLAIM&displayToyaMA=displayToyaMA&displayOSU=displayOSU&displayRecomb=displayRecomb&displayLisa=displayLisa&"
+    "displayRFI=displayRFI&ls1=ls1&ls2=ls2&ls3=ls3&ls4=ls4&ls5=ls5&el1=el1&el2=el2&el3=el3&el4=el4&show_unres_qn=show_unres_qn&"
+    "submit=Export&export_type=current&export_delimiter=tab&offset=0&limit=100000&range=on";
+
+SpectralLineRequest::SpectralLineRequest() {}
+
+SpectralLineRequest::~SpectralLineRequest() {}
+
+/* Referenced from curl example https://curl.haxx.se/libcurl/c/getinmemory.html */
+void SpectralLineRequest::SendRequest(const CARTA::DoubleBounds& frequencyRange, CARTA::SpectralLineResponse& spectral_line_response) {
```

Comment:
I think [this approach](https://gist.github.com/alghanmi/c5d7b761b2c9ab199157#file-curl_example-cpp) looks a little more C++-friendly. I'd like to avoid `malloc`, `realloc` and `free` if possible